src/functionaltests/java/com/ericsson/ei/utils/HttpDeleteRequest.java

```diff
@@ -1,34 +1,14 @@
 package com.ericsson.ei.utils;
 
-import com.ericsson.ei.controller.model.SubscriptionResponse;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import lombok.Getter;
-import lombok.Setter;
-import lombok.experimental.Accessors;
 import org.apache.http.client.methods.HttpDelete;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.springframework.http.ResponseEntity;
 
-import java.io.IOException;
-
-
-@Accessors(chain = true)
-public class HttpDeleteRequest {
-    private static final Logger LOGGER = LoggerFactory.getLogger(HttpDeleteRequest.class);
-    @Getter @Setter private int port;
-    @Getter @Setter private String url;
-    @Getter @Setter private String endpoint;
-    private ObjectMapper mapper = new ObjectMapper();
-    private SubscriptionRestAPI restApi = new SubscriptionRestAPI();
-
+public class HttpDeleteRequest extends HttpRequest {
 
     /**
      * Build together a httpDelete object
-     * */
-    public ResponseEntity<String>  build() {
-        HttpDelete httpDelete = new HttpDelete(url + port + endpoint);
-
-        return restApi.getResponse(httpDelete);
+     */
+    public ResponseEntity<String> build() {
```

Comment:
Change functional name on build() function to something more valid name, since actual also do the request. Maybe change to name "performRequest()" or something. Same comment for all HTTP Request classes.
Update the java doc so it describes more what it do.

Reply:
build is necessary when implementing build pattern. However do request should not be done here. It should however be a new story to refactor these classes.

Reply:
I didn't see your comment until now. I remade the way HttpRequest works without split classes. The thing is that when I started to split up the test methods I noticed that they weren't modular. I needed to be able to add a credentials header for example to GET or POST request without knowing which method it is. That is why I made the parent class HttpRequest and then I began to see how much they had in common and that the only thing that split them appart was one line. Should i revert back and find another solution?